packages/vx-event/src/typeGuards.ts

```diff
@@ -0,0 +1,27 @@
+export function isElement(elem?: Element | Event): elem is Element {
+  return !!elem && elem instanceof Element;
+}
+
+// functional definition of isSVGElement. Note that SVGSVGElements are HTMLElements
+export function isSVGElement(elem?: Element): elem is SVGElement {
+  return !!elem && (elem instanceof SVGElement || 'ownerSVGElement' in elem);
+}
+
+// functional definition of SVGGElement
+export function isSVGSVGElement(elem?: Element | null): elem is SVGSVGElement {
+  return !!elem && 'createSVGPoint' in elem;
+}
+
+export function isSVGGraphicsElement(elem?: Element | null): elem is SVGGraphicsElement {
+  return !!elem && 'getScreenCTM' in elem;
+}
+
+// functional definition of TouchEvent
+export function isTouchEvent(event?: Event): event is TouchEvent {
+  return !!event && 'changedTouches' in event;
+}
+
+// functional definition of event
+export function isEvent(event?: Event | Element): event is Event {
+  return !!event && 'target' in event;
```

Comment:
`event instanceof Event`?

Reply:
I wasn't 100% sure if that would play well with `React` `events`, I'll give it a try functionally and update if it works 👍

Reply:
looks like `event instanceof Event` is false for `React` `events`

Reply:
Yeah React events are `SyntheticEvent`, which have a property `nativeEvent`, that points to the actual `Event`.

Reply:
👍 I added `@types/react` to this package so we could support `React.MouseEvent | React.TouchEvent`, and updated this check to include `instance of` on either the passed `event` or `event.nativeEvent`. Seems like a good solution.